### PR TITLE
boards: shields: add support for buydisplay 2.8" TFT touch display 

### PR DESCRIPTION
**Pull request by gmarull**
Add support for the Buydisplay 2.8" TFT touch display shield for Arduino.

### CODE REVIEW

File: boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts [hunk 1]
```diff
@@ -188,7 +188,7 @@ arduino_i2c: &i2c0 {
 
 &spi1 {
 	compatible = "nordic,nrf-spi";
-	status = "okay";
+	status = "disabled";
```
Comment (gmarull) [inline boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts:191]: This shield uses `gpio1.8` (`miso-pin = <40>`), unless SPI1 is disabled it doesn't work. I'm not sure if there is a better way to solve that...

Comment (jfischer-phytec-iot) [reply boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts:191]: I think the only way is to add `boards/shields/buydisplay_2_8_tft_touch_arduino/boards/nrf52840dk_nrf52840.overlay` and disable spi1 node there

Comment (gmarull) [reply boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts]: @jfischer-phytec-iot ok, it works. thanks for the suggestion!

**Review by jfischer-no — APPROVED** [review]

**Review by vanwinkeljan — APPROVED** [review]

**Comment by carlescufi** [comment]
@gmarull please rebase

**Comment by gmarull** [comment]
@carlescufi done

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**